lib/payload/script-engine.vala

```diff
@@ -432,10 +432,15 @@ namespace Frida {
 			}
 
 			public void post (string message, Bytes? data) throws Error {
-				if (state != LOADED && state != DISPOSED && state != LOADING)
-					throw new Error.INVALID_OPERATION ("Only loading/loaded scripts may be posted to");
-
-				script.post (message, data);
+				switch(state) {
+					case LOADING:
+					case LOADED:
+					case DISPOSED:
+					throw new Error.INVALID_OPERATION ("Only scripts still alive may be posted to");
```

Comment:
```suggestion
						throw new Error.INVALID_OPERATION ("Only active scripts may be posted to");
```

(Fixing indent and making it clearer for the case where somebody forgets to call `.load ()`.)

Reply:
Or perhaps: `Only live scripts may be posted to`